clients/projectshare/src/ProjectShareClient.ts

```diff
@@ -524,7 +524,7 @@ export class ProjectShareClient extends WsgClient {
     try {
       const accessurl = file.accessUrl !== undefined ? file.accessUrl : "";
       await this.uploadBlob(requestContext, accessurl, data);
-      return this.updateFileExistsProperty(requestContext, contextId, file.wsgId);
+      return await this.updateFileExistsProperty(requestContext, contextId, file.wsgId);
```

Comment:
You should not have to await the return of a promise here. The Promise itself should be returned and then awaited by the caller of the `uploadContentInFile()`. Were you seeing an issue with it?

Reply:
I was not seeing an issue with it. This is an automatic fix concluded by the new lint rule I introduced.
The two reasons to do it this way according to the [newly used linter rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md).
- I believe in this particular case the goal is that the `catch` will handle the promise being rejected, rather than letting the caller have to handle that when it awaits it.
- allegedly better async call stacks

Reply:
it is also important to note (as I did in the PR description but it's long so worth mentioning here), that there may be a *lot* more autofixes incoming from this rule. The rule is actually quite broken (it fails to handle any cases with nested functions), but there is a PR to fix that incoming which should eventually trigger this linter error in more places.

Reply:
@calebmshafer do you have any resolution preference here? Would you prefer I disable the linter for this one and restore the original behavior? I think in this case only minor scheduling will be different since the catch block will just rethrow to the caller anyway, and I think we should keep the linter rule's recommendation.

Reply:
fair enough, that makes sense. I assume since you turned on the rule the rest of the PR covers all of the places required?